airbyte-integrations/bases/base-java/src/main/java/io/airbyte/integrations/destination_async/RunningFlushWorkers.java

```diff
@@ -12,16 +12,24 @@
 import java.util.UUID;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.concurrent.ConcurrentMap;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import lombok.extern.slf4j.Slf4j;
 
 /**
  * Track the number of flush workers (and their size) that are currently running for a given stream.
  */
+@Slf4j
 public class RunningFlushWorkers {
 
   private final ConcurrentMap<StreamDescriptor, ConcurrentMap<UUID, Optional<Long>>> streamToFlushWorkerToBatchSize;
+  private final ScheduledExecutorService debugLoop;
 
   public RunningFlushWorkers() {
     streamToFlushWorkerToBatchSize = new ConcurrentHashMap<>();
+    debugLoop = Executors.newSingleThreadScheduledExecutor();
```

Comment:
let's make this a daemon thread
```suggestion
    debugLoop = ExecutorService exec = Executors.newFixedThreadPool(1,
        new ThreadFactory() {
            public Thread newThread(Runnable r) {
                Thread t = Executors.defaultThreadFactory().newThread(r);
                t.setDaemon(true);
                return t;
            }
        });
```

then you don't need to explicitly shut down the pool